### PR TITLE
Update Ordering, wording and help text for private pages

### LINKED ISSUE #11535: Update wording and help text for private pages using shared password
### Is your proposal related to a problem?

Wagtail's page privacy feature could be improved to communicate better to the user that the password created is different from using authentication (log in) to restrict access.

This issue is one part of a wider plan to improve this feature - see #10588

### Describe the solution you'd like

For private access dialogs (collection and page privacy), the following should be changed;

1. Update the ordering of the private access so that it goes from most public to most secure. 
  * The current ordering with the 'shared password' between the two logged in options does not make it easy to communicate the differences.
2. Change the wording for the labels:
  * "Private, accessible with the following password" -> "Private, accessible with a shared password"
  * "Private, accessible to logged-in users" -> "Private, accessible to any logged-in users"
3. Update the field label from "Password" to "Shared password"
4. Add help text (use our shared field template includes for this) with help text as follows:
  * "Shared passwords should not be used to protect sensitive content, anyone who has this password will be able to view the content."
5. Ensure all other references for private pages/collections are updated with references to 'password' becoming 'shared password'. Such as;
  * Documentation - https://docs.wagtail.org/en/stable/advanced_topics/privacy.html & https://docs.wagtail.org/en/stable/topics/permissions.html#image-document-permissions
  * Audit logging
  * Selected state in the side panel or the collection edit page
6. [ ] Once merged - a new issue on the Wagtail Guide will be needed to ensure screenshots get updated https://guide.wagtail.org/en-latest/how-to-guides/manage-collections/#privacy-settings

#### Screenshots of current state

Screenshot 2024-01-26 at 4 51 17 pm

Screenshot 2024-01-26 at 4 49 44 pm

Audit logging

247612418-69278dd0-028d-4e69-a81e-493d8e161367

### Describe alternatives you've considered

There were various discussions on https://github.com/wagtail/wagtail/issues/10588 about other approaches, the core team have agreed that this is a good first step.

### Additional context

* Some of this (help text) was done in https://github.com/wagtail/wagtail/pull/10729 - but do not use this PR, instead create a new one.
* Documentation - https://docs.wagtail.org/en/stable/advanced_topics/privacy.html & https://docs.wagtail.org/en/stable/topics/permissions.html#image-document-permissions

#### Out of scope

Updating the default password template form does not need to be changed. Many sites will customise this.

Screenshot 2024-01-26 at 4 52 14 pm

### Working on this

* Anyone can contribute to this who is willing to take the time to understand how the private access features work and review the changes suggested above and work through the code to find out where to update this.
* Unit tests and documentation must be updated with any PR
* View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

### PR DESCRIPTION
**Pull request by rohitsrma**
Fixes #11535

### After Changes:

image

**Help Text**

image

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by rohitsrma** [comment]
@lb- I would greatly appreciate your feedback and suggestions on the current changes. I'll update the tests once I am finished with the updates and everything looks satisfactory.

**Comment by lb-** [comment]
Sorry it was not clear on the issue, can you please put the shared password option under 'public'.

File: wagtail/models/view_restrictions.py [hunk 1]
```diff
@@ -19,13 +19,20 @@ class BaseViewRestriction(models.Model):
...
     )
 
     restriction_type = models.CharField(max_length=20, choices=RESTRICTION_CHOICES)
-    password = models.CharField(verbose_name=_("password"), max_length=255, blank=True)
+    password = models.CharField(
+        verbose_name=_("shared password"),
+        max_length=255,
+        blank=True,
+        help_text=_(
+            "Shared passwords should not be used to protect sensitive content. Anyone who has this password will be able to view the content."
```
Comment (lb-) [inline wagtail/models/view_restrictions.py:33]: Can you confirm this help text shows in the UI. Just add a screenshot.

Comment (rohitsrma) [reply wagtail/models/view_restrictions.py:33]: Done!

**Comment by rohitsrma** [comment]
Thanks @lb-, I've made the changes

**Review by lb- — APPROVED** [review]
Looks great, tested a few scenarios and audit logs.

Thank you so much @rohitsrma 

Screenshot 2024-01-30 at 7 34 54 pm

Screenshot 2024-01-30 at 7 34 06 pm

**Pull request merged**